### ISSUE TITLE
Requires old version of setuptools-git

### ISSUE THREAD

**Issue by rhertzog** [acsone/setuptools-odoo#23]
From setup.py:
```python
    install_requires=[
        'setuptools',
        'setuptools-git!=1.2',
    ],
```

However setuptools-git's latest version is 1.2 (https://pypi.python.org/pypi/setuptools-git). This kind of requirement make it impossible to create a Debian package for setuptools-odoo because Debian already contains version 1.2 of setuptools-git and it's not going to switch to an older version...

Instead of requiring an old version, you should:
- either work with setuptools-git to make a new version that is usable for you
- or you should embed the logic that you need in setuptools-odoo and get rid of the dependency altogether
- or find another way to achieve the desired effect

/cc @sbrun

**Comment by sbidoul** [comment]
There is an open PR on setuptools-git: https://github.com/msabramo/setuptools-git/pull/13 . You can try giving it a gentle push and ask for a release. setuptools-git seems to be unmaintained though. 

The best approach is to add the missing symlink support to https://github.com/pypa/setuptools_scm. It should then be a drop-in replacement for setuptools-git.

**Comment by sbidoul** [comment]
Note that setuptools-odoo is not required to deploy or run built packages.

odoo-autodiscover is required at runtime for Odoo 8, 9 and 10. That one might be a tad harder to package due to it's reliance on a .pth file to do its magic.

**Comment by rhertzog** [comment]
Thanks for the advice. We are only targetting Odoo 11 so I think that odoo-autodiscover is not required at all, correct?

**Comment by sbidoul** [comment]
That is correct.

**Comment by rhertzog** [comment]
Well, we can package it and ignore the version mismatch in Debian. Would that result in a problematic behaviour when used in setup.py contained in release tarballs? Those tarballs have no .git directory so I guess that they are not causing any trouble and that this issue should not be a blocker.

**Comment by rhertzog** [comment]
So maybe you want to add code to detect that situation and bail out?

Or monkey patch the problematic module with a fixed version that you are providing?

But on my side, there's nothing I can do to replace the new version by the old one.

**Comment by sbidoul** [comment]
My question was if you can avoid distributing setuptools-odoo.deb and only use it for packaging addons.

Anyway I'll see what it involves to implement proper support for symlinks in setuptools_scm.

**Comment by sbidoul** [comment]
@rhertzog actually I think you don't need to package setuptools-odoo at all. Since it's referenced in setup_requires, setuptools will download it (and it's setuptools-git dependency) from pypi on demand and only when packaging, not at runtime.

**Comment by rhertzog** [comment]
@sbidoul The Debian policy doesn't allow network access during package build and code used during build is supposed to be available in Debian too, so it's not really an option if we want to integrate our packages in Debian.

**Comment by sbidoul** [comment]
@rhertzog Done in [version 2.3.0](https://pypi.org/project/setuptools-odoo/), which now depends on the just released [setuptools_scm 2.1.0](https://pypi.org/project/setuptools_scm/2.1.0/) instead of setuptools-git.
